ckan/tests/helpers.py

```diff
@@ -468,3 +544,149 @@ def wrapper(*args, **kwargs):
+
+        logs.assert_log(u'info', u'world')
+    '''
+    if logger is None:
+        logger = logging.getLogger()
+    elif not isinstance(logger, logging.Logger):
+        logger = logging.getLogger(logger)
+    handler = RecordingLogHandler()
+    old_level = logger.level
+    manager_level = logger.manager.disable
+    disabled = logger.disabled
+    logger.addHandler(handler)
+    try:
+        if level is not None:
+            logger.setLevel(level)
+        if override_disabled:
+            logger.disabled = False
+        if override_global_level:
+            if (level is None) and (manager_level > old_level):
+                logger.manager.disable = old_level
+            elif (level is not None) and (manager_level > level):
+                logger.manager.disable = level
+        yield handler
+    finally:
+        logger.handlers.remove(handler)
+        logger.setLevel(old_level)
+        logger.disabled = disabled
+        logger.manager.disable = manager_level
+
+
+class RecordingLogHandler(logging.Handler):
+    u'''
+    Log handler that records log messages for later inspection.
+
+    You can inspect the recorded messages via the ``messages`` attribute
+    (a dict that maps log levels to lists of messages) or by using
+    ``assert_log``.
+    '''
+    def __init__(self, *args, **kwargs):
+        super(RecordingLogHandler, self).__init__(*args, **kwargs)
+        self.clear()
+
+    def emit(self, record):
+        self.messages[record.levelname.lower()].append(record.getMessage())
+
+    def assert_log(self, level, pattern, msg=None):
+        u'''
+        Assert that a certain message has been logged.
+
+        :param string pattern: A regex which the message has to match.
+            The match is done using ``re.search``.
+
+        :param string level: The message level (``'debug'``, ...).
+
+        :param string msg: Optional failure message in case the expected
+            log message was not logged.
+
+        :raises AssertionError: If the expected message was not logged.
+        '''
+        compiled_pattern = re.compile(pattern)
+        for log_msg in self.messages[level]:
+            if compiled_pattern.search(log_msg):
+                return
+        if not msg:
+            if self.messages[level]:
+                lines = u'\n    '.join(self.messages[level])
+                msg = (u'Pattern "{}" was not found in the log messages for '
+                       + u'level "{}":\n    {}').format(pattern, level, lines)
+            else:
+                msg = (u'Pattern "{}" was not found in the log messages for '
+                       + u'level "{}" (no messages were recorded for that '
+                       + u'level).').format(pattern, level)
+        raise AssertionError(msg)
+
+    def clear(self):
+        u'''
+        Clear all captured log messages.
+        '''
+        self.messages = collections.defaultdict(list)
+
+
+@contextlib.contextmanager
+def temp_file(*args, **kwargs):
```

Comment:
I think you're right. Usually my main problem with `NamedTemporaryFile` is that it deletes the file when it's being closed instead of when the context manager is left (which would often be more useful for me). So I end up passing `delete=False` to `NamedTemporaryFile` and do the cleanup myself -- that's what `temp_file` does (I've copied it from other projects where it's been useful).

However, I just checked my usages of `temp_file` in this PR and it seems that I don't need that modified deletion behavior at all. So, yes, I think I can remove it.